includes/Services/GitContentRepoManager.php

```diff
@@ -0,0 +1,194 @@
+<?php
+
+declare(strict_types=1);
+
+namespace LabkiPackManager\Services;
+
+use LabkiPackManager\Util\UrlResolver;
+use MediaWiki\MediaWikiServices;
+use RuntimeException;
+
+/**
+ * GitContentRepoManager
+ *
+ * Manages local git repository clones for content repositories.
+ * Handles cloning, updating, and ensuring repositories are available locally
+ * for specific branches, tags, or commits.
+ */
+final class GitContentRepoManager {
+    private string $cloneBasePath;
+    private LabkiRepoRegistry $repoRegistry;
+
+    public function __construct(?LabkiRepoRegistry $repoRegistry = null) {
+        global $wgCacheDirectory;
+        $this->cloneBasePath = $wgCacheDirectory
+            ? "{$wgCacheDirectory}/labki-content-repos"
+            : wfTempDir() . '/labki-content-repos';
```

Comment:
is this introducing indeterminacy in the repo directory? i can't quite tell, but we want this to be deterministic, either absolute path in the database or a single root path that's true or else raises